fe/fe-core/src/main/java/org/apache/doris/planner/external/IcebergScanProvider.java

```diff
@@ -168,6 +176,18 @@ public List<InputSplit> getSplits(List<Expr> exprs) throws UserException {
 
         org.apache.iceberg.Table table = getIcebergTable();
         TableScan scan = table.newScan();
+        TableSnapshot tableSnapshot = desc.getRef().getTableSnapshot();
+        if (tableSnapshot != null) {
+            TableSnapshot.VersionType type = tableSnapshot.getType();
+            if (type == TableSnapshot.VersionType.VERSION) {
+                scan = scan.useSnapshot(tableSnapshot.getVersion());
```

Comment:
What if user give a non-exist snapshot version?

Reply:
Catch the `NotFoundException` and throw `UserException`

Reply:
catch illegalArgument